ext/rotoscope/rotoscope.c

```diff
@@ -58,35 +59,27 @@ static bool is_class_singleton(VALUE klass) {
 
 static VALUE singleton2str(VALUE klass) {
   if (is_class_singleton(klass)) {
-    VALUE obj = class_of_singleton(klass);
-    VALUE cached_lookup = rb_class_path_cached(obj);
-    VALUE name = (NIL_P(cached_lookup)) ? rb_class_name(obj) : cached_lookup;
-    return name;
+    return class2str(class_of_singleton(klass));
   } else  // singleton of an instance
   {
     return singleton2str(CLASS_OF(klass));
   }
 }
 
-static rs_class_desc_t class2str(VALUE klass) {
-  rs_class_desc_t real_class;
-  real_class.method_level = INSTANCE_METHOD;
+static VALUE class_path(VALUE klass) {
+  VALUE cached_path = rb_class_path_cached(klass);
+  if (!NIL_P(cached_path)) {
+    return cached_path;
+  }
+  return rb_class_path(klass);
+}
```

Comment:
Extracted this function out since we should always do these together.  We shouldn't return something different based on whether or not the class path is cached, which the code was doing before the commit I just added.  We should also always try to get the class path from `rb_class_path_cached` first, since `rb_class_path` duplicates the string.